### PR TITLE
[plugins][scsi] Enhance data gathering of tape devices

### PR DESCRIPTION
**Pull request by jcastill**
- Add a function to iterate through tape devices, similar to
_get_block_devs() and _get_fibre_devs(), and
- Capture udevadm info for tape devices. 

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [X] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [X] Is the subject and message clear and concise?
- [X] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [X] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [X] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?

### CODE REVIEW

**Comment by jcastill** [comment]
Draft PR for now because I'm still testing this. 
I'll try to get reservation key information as well but it's a bit trickier so far.

**Comment by jcastill** [comment]
At this point I wonder if it wouldn't be better to have a 'tape' or 'tapes' plugin by itself, like we have for block devices...

File: sos/report/__init__.py [hunk 1]
```diff
@@ -506,6 +507,26 @@ def _get_block_devs(self):
...
+        These devices are used by add_device_cmd() in the Plugin class.
+        """
+        try:
+            devs = []
+            devdirs = [
+                'scsi_tape',
+                'lin_tape'
+            ]
+            for devdir in devdirs:
+                if os.path.isdir(f"/sys/class/{devdir}"):
```
Comment (pmoravec) [inline sos/report/__init__.py:523]: This is redundant test:

    >>> os.path.isdir("/blah")
    False
    >>> glob.glob("/blah")
    []
    >>> 

(purely from performance point of view, I even did a test comparing both approaches. It seems Python spends bit more time in `isdir` test every time than in running extra `glob` and adding empty list sometimes)

**Review by pmoravec — APPROVED** [review]

File: sos/report/__init__.py [hunk 2]
```diff
@@ -506,6 +507,25 @@ def _get_block_devs(self):
...
+        """Enumerate a list of tape devices on this system so that plugins
+        can iterate over them
+
+        These devices are used by add_device_cmd() in the Plugin class.
+        """
+        try:
+            devs = []
+            devdirs = [
+                'scsi_tape',
+                'lin_tape'
```
Comment (arif-ali) [inline sos/report/__init__.py:520]: minor nit, comma at the end. But all good all round

**Review by arif-ali — APPROVED** [review]

File: sos/report/__init__.py (hunk 2 above)
Comment (pmoravec) [reply sos/report/__init__.py:520]: I will merge the PR and add the coma to it.

**Comment by arif-ali** [comment]
@jcastill did you need to do anything further on this, both me and @pmoravec thought this was ok?

**Comment by pmoravec** [comment]
Hi @jcastill , is there a reason to keep this as a Draft? If not, we can merge it, PR got 2 approvals.

**Comment by jcastill** [comment]
Thank you @arif-ali and @pmoravec . This is still a draft because I've been waiting to run some tests in the lab with the storage guys. I'll see if I can get some hardware next week and will aim to have this ready the week after.

**Comment by jcastill** [comment]
I'm going to have to mock a system with tape devices because I cannot get hold of the real thing. I'll update soon with the test results.

**Comment by arif-ali** [comment]
@jcastill did you have any luck with this?

**Comment by jcastill** [comment]
@arif-ali no luck, but this has been waiting on me/hardware requests for a bit. So I'm going to close it for now, then work on an environment with scsi_debug or mhvtl.
